lib/db_connection/connection.ex

```diff
@@ -169,23 +169,25 @@ defmodule DBConnection.Connection do
         mon = Process.monitor(pid)
         handle_checkout({ref, mon}, timeout, from, state, s)
       %{client: :closed} ->
-        err = DBConnection.Error.exception("connection not available")
+        message = "connection not available because of disconnection"
+        err = DBConnection.Error.exception(message)
         {:reply, {:error, err}, s}
       %{queue: queue} when queue? == true ->
         client = {ref, Process.monitor(pid)}
         queue = :queue.in({client, timeout, from}, queue)
         {:noreply, %{s | queue: queue}}
       _ when queue? == false ->
-        err = DBConnection.Error.exception("connection not available")
+        message = "connection not available because of queue"
```

Comment:
"connection not immediately available"?

On 27 April 2016 at 06:15, José Valim <notifications@github.com> wrote:

> In lib/db_connection/connection.ex
> <https://github.com/fishcakez/db_connection/pull/34#discussion_r61204857>:
>
> >          {:reply, {:error, err}, s}
> >        %{queue: queue} when queue? == true ->
> >          client = {ref, Process.monitor(pid)}
> >          queue = :queue.in({client, timeout, from}, queue)
> >          {:noreply, %{s | queue: queue}}
> >        _ when queue? == false ->
> > -        err = DBConnection.Error.exception("connection not available")
> > +        message = "connection not available because of queue"
>
> "because of queue"?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/fishcakez/db_connection/pull/34/files/7a9fef74d3ea8983644a3c201777ce25d89b3b39#r61204857>
>